core/src/test/java/org/elasticsearch/common/xcontent/BaseXContentTestCase.java

```diff
@@ -182,4 +824,71 @@ protected void doTestBigInteger(JsonGenerator generator, ByteArrayOutputStream o
         assertEquals("bar", map.get("foo"));
         assertEquals(bigInteger, map.get("bigint"));
     }
+
+    public void testEnsureNameNotNull() {
+        IllegalArgumentException e = expectThrows(IllegalArgumentException.class, () -> XContentBuilder.ensureNameNotNull(null));
+        assertThat(e.getMessage(), containsString("Field name cannot be null"));
+    }
+
+    public void testFormatterNameNotNull() {
+        IllegalArgumentException e = expectThrows(IllegalArgumentException.class, () -> XContentBuilder.ensureFormatterNotNull(null));
+        assertThat(e.getMessage(), containsString("DateTimeFormatter cannot be null"));
+    }
+
+    public void testEnsureNotNull() {
+        IllegalArgumentException e = expectThrows(IllegalArgumentException.class, () -> XContentBuilder.ensureNotNull(null, "message"));
+        assertThat(e.getMessage(), containsString("message"));
+
+        XContentBuilder.ensureNotNull("foo", "No exception must be thrown");
+    }
+
+    private static void expectUnclosedException(ThrowingRunnable runnable) {
+        IllegalStateException e = expectThrows(IllegalStateException.class, runnable);
+        assertThat(e.getMessage(), containsString("Failed to close the XContentBuilder"));
+        assertThat(e.getCause(), allOf(notNullValue(), instanceOf(IOException.class)));
+        assertThat(e.getCause().getMessage(), containsString("Unclosed object or array found"));
+    }
+
+    private static void expectValueException(ThrowingRunnable runnable) {
+        JsonGenerationException e = expectThrows(JsonGenerationException.class, runnable);
+        assertThat(e.getMessage(), containsString("expecting a value"));
+    }
+
+    private static void expectFieldException(ThrowingRunnable runnable) {
+        JsonGenerationException e = expectThrows(JsonGenerationException.class, runnable);
+        assertThat(e.getMessage(), containsString("expecting field name"));
+    }
+
+    private static void expectNonNullFieldException(ThrowingRunnable runnable) {
+        IllegalArgumentException e = expectThrows(IllegalArgumentException.class, runnable);
+        assertThat(e.getMessage(), containsString("Field name cannot be null"));
+    }
+
+    private static void expectObjectException(ThrowingRunnable runnable) {
+        JsonGenerationException e = expectThrows(JsonGenerationException.class, runnable);
+        assertThat(e.getMessage(), containsString("Current context not Object"));
+    }
+
+    private static void expectArrayException(ThrowingRunnable runnable) {
+        JsonGenerationException e = expectThrows(JsonGenerationException.class, runnable);
+        assertThat(e.getMessage(), containsString("Current context not Array"));
+    }
+
+    public static org.hamcrest.Matcher<String> equalToJSon(String json) {
+        return org.hamcrest.core.IsEqual.equalTo(json.replace("'", "\""));
```

Comment:
`Matchers.equalTo` or `equalTo` with a static import is more normal for this sort of thing I think.